### ISSUE TITLE
auto stash uncommited changes when doing rr and auto pop stashed changes when done with rr

### ISSUE THREAD

**Issue by drewdeponte** [uptech/git-ps#39]

**Comment by drewdeponte** [comment]
Thinking about this some more this should be an optional switch to control this.

**Comment by ctsstc** [comment]
If there are changes maybe a prompt asking if should:
- stash
- cancel

Maybe a third option like: 
- append changes to last commit
- append changes to requested commit at index: [index]

**Comment by brandonpittman** [comment]
I'd be happy to just have the `rr` work as normal and then pop the unstaged files back in. The issue I always run into is that I'll want to `rr` some commits, but I did some work before I requested the review. It's never that I want to append the current working files onto existing commits.

**Comment by drewdeponte** [comment]
Yeah I am in the camp with @brandonpittman i think we just need a switch we can provide on the command line to control if it auto stashes and pops or not.

Maybe something like

`git-ps rr -s <patch-index>`

Short for stash, or maybe

`git-ps rr -a <patch-index>`

Short for auto-stash.

Then if people prefer one behavior over the other they can control it with aliases.

**Comment by brandonpittman** [comment]
`-s` seems good to me!
